src/bid/encoding.rs

```diff
@@ -63,50 +64,168 @@ impl Into<StorageScalar> for &StorageBid {
         // Byte-types are treated in Little Endian.
 
         // Safe unwrap here.
-        let type_fields = Scalar::from_bytes(b"44223133000000000000000000000000").unwrap();
+        let type_fields =
+            Scalar::from_bytes(b"44223133000000000000000000000000").unwrap();
         words_deposit.push(type_fields);
 
         // 2. Encode each word.
-        // Scalar and any other type that can be embedded in, will also be treated as one.
+        // Scalar and any other type that can be embedded in, will also be
+        // treated as one.
         words_deposit.push(Scalar::from(self.elegibility_ts as u64));
         words_deposit.push(Scalar::from(self.expiration_ts as u64));
-        // Unwraping a conversion between JubJubScalar to BlsScalar is always safe since the order of
-        // the JubJubScalar field is shorter than the BlsScalar one.
-        words_deposit.push(Scalar::from_bytes(&self.encrypted_blinder.to_bytes()).unwrap());
-        words_deposit.push(Scalar::from_bytes(&self.encrypted_value.to_bytes()).unwrap());
+
+        // A conversion between JubJubScalar to BlsScalar is always safe since
+        // the order of a JubJubScalar field is shorter than a BlsScalar one.
+        words_deposit.push(jubjub_scalar_to_bls12_381(self.encrypted_blinder));
+        words_deposit.push(jubjub_scalar_to_bls12_381(self.encrypted_value));
+
         // Push both JubJubAffine coordinates as a Scalar.
-        words_deposit.push(Scalar::from_bytes(&self.randomness.get_x().to_bytes()).unwrap());
-        words_deposit.push(Scalar::from_bytes(&self.randomness.get_y().to_bytes()).unwrap());
+        words_deposit.push(self.randomness.get_x());
```

Comment:
I'm a bit confused about this part. Seems the original intent was to grant the scalar belongs to `Fq`, but this is implicit considering any x or y from `JubJub` are automatically in `Fq`. Is this assumption correct? If yes, then this change is valid. We should check with someone from the theoretical part.

Reply:
If I understand the question, yes. The `x` and `y` from `JubJub` are, in fact, `Fq`.